### PR TITLE
[CR-215] Resolve dependency conflicts with KSCrash 

### PR DESCRIPTION
**Pull request by mduncan26**
This PR is intended to solve the compilation issues when raygun4apple is integrated through CocoaPods. Other third party libraries can use either KSCrash or include header/source files that cause compilation issues. This is due to files/classes/methods being named the same in different files which the compiler doesn't know which definition to use.

The solution is to add a prefix so that the KSCrash files are unique to Raygun.

Changes:
- Prefix files with Raygun
- Prefix classes and methods with Raygun
- Prefix macros with Raygun

### CODE REVIEW

File: Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor.c [hunk 1]
```diff
@@ -137,52 +137,52 @@ static inline void setMonitorEnabled(Monitor* monitor, bool isEnabled)
...
         }
-        monitorTypes &= KSCrashMonitorTypeDebuggerSafe;
+        monitorTypes &= Raygun_KSCrashMonitorTypeDebuggerSafe;
     }
-    if(g_requiresAsyncSafety && (monitorTypes & KSCrashMonitorTypeAsyncUnsafe))
+    if(g_requiresAsyncSafety && (monitorTypes & Raygun_KSCrashMonitorTypeAsyncUnsafe))
     {
-        KSLOG_DEBUG("Async-safe environment detected. Masking out unsafe monitors.");
-        monitorTypes &= KSCrashMonitorTypeAsyncSafe;
+        RAYGUN_KSLOG_ERROR("Async-safe environment detected. Masking out unsafe monitors.");
```
Comment (QuantumNightmare) [inline Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor.c:179]: I see several log statements here that have changed from DEBUG/INFO to ERROR, are all these intentional?

Comment (mduncan26) [reply Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor.c]: They weren't intentional. Must of been a quick find and replace mistake. I have gone back though and reverted them.

Comment (QuantumNightmare) [reply Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor.c]: Thanks, I'll continue my review

**Review by MattByers — APPROVED** [review]
Looks good as far as I can tell. Didn't come across any obvious typos or copy paste errors.

File: Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor_MachException.c [hunk 2]
```diff
@@ -291,60 +291,60 @@ static void* handleExceptions(void* const userData)
...
-        KSLOG_DEBUG("Exception handler is installed. Continuing exception handling.");
+        RAYGUN_KSLOG_DEBUG("Exception handler is installed. Continuing exception handling.");
 
 
         // Switch to the secondary thread if necessary, or uninstall the handler
         // to avoid a death loop.
-        if(ksthread_self() == g_primaryMachThread)
+        if(raygun_ksthread_self() == g_primaryMachThread)
         {
-            KSLOG_DEBUG("This is the primary exception thread. Activating secondary thread.");
```
Comment (QuantumNightmare) [inline Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor_MachException.c:312]: Debug->Error, was this one intentional?

File: Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+// ============================================================================
+
+/** Called when a crash occurs.
+ *
+ * This function gets passed as a callback to a crash handler.
+ */
+static void onCrash(struct Raygun_KSCrash_MonitorContext* monitorContext)
+{
+    if (monitorContext->currentSnapshotUserReported == false) {
+        RAYGUN_KSLOG_ERROR("Updating application state to note crash.");
```
Comment (QuantumNightmare) [inline Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c:97]: Debug changed to Error, looks unintentional here

Comment (QuantumNightmare) [reply Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c:97]: And 3 other places in this file

**Review by samuel-holt — APPROVED** [review]
Changes look good :+1: Objective-C breaks my mind 🤯

File: Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor_MachException.c (hunk 2 above)
Comment (mduncan26) [reply Sources/Raygun_KSCrash/Recording/Monitors/Raygun_KSCrashMonitor_MachException.c:312]: Not intentional, I have reverted it.

File: Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c (hunk 3 above)
Comment (mduncan26) [reply Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c]: Good catch I have fixed them up.

File: Sources/Raygun_KSCrash/llvm/Support/Raygun_Compiler.h [hunk 4]
```diff
@@ -12,10 +12,10 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef LLVM_SUPPORT_COMPILER_H
-#define LLVM_SUPPORT_COMPILER_H
+#ifndef RAYGUN_LLVM_SUPPORT_COMPILER_H
+#define RAYGUN_LLVM_SUPPORT_COMPILER_H
 
-#include "llvm-config.h"
+#include "Raygun_llvm-config.h"
```
[the remark this answers is not in the document]
Comment (mduncan26) [reply Sources/Raygun_KSCrash/llvm/Support/Raygun_Compiler.h:18]: It's based on the filename which is capitalised.

File: Sources/Raygun_KSCrash/llvm/Support/Raygun_Casting.h [hunk 5]
```diff
@@ -12,11 +12,11 @@
...
 
-#ifndef LLVM_SUPPORT_CASTING_H
-#define LLVM_SUPPORT_CASTING_H
+#ifndef RAYGUN_LLVM_SUPPORT_CASTING_H
+#define RAYGUN_LLVM_SUPPORT_CASTING_H
 
-#include "Compiler.h"
-#include "type_traits.h"
+#include "Raygun_Compiler.h"
+#include "Raygun_type_traits.h"
```
[the remark this answers is not in the document]
Comment (mduncan26) [reply Sources/Raygun_KSCrash/llvm/Support/Raygun_Casting.h:19]: It's based on the filename which is capitalised.

**Review by QuantumNightmare — APPROVED** [review]
Looks good :+1:

**Pull request merged**